commands/push.go

```diff
@@ -141,3 +150,30 @@ func validateImages(functions map[string]stack.Function) []string {
 	}
 	return invalidImages
 }
+
+func overrideImagePrefix(yamlImageName string) string {
+	if yamlImageName == "" {
+		return ""
+	}
+
+	imageNameParts := strings.Split(yamlImageName, "/")
+	var (
+		yamlPrefix    string
+		yamlHasPrefix = (len(imageNameParts) > 1)
+	)
+	if yamlHasPrefix {
+		// prefix is set in yaml
+		yamlPrefix = imageNameParts[0]
+	}
+
+	imagePrefix = getImagePrefix(imagePrefix, os.Getenv(imagePrefixEnvironment), yamlPrefix)
+	if yamlHasPrefix {
+		// update prefix in place
+		imageNameParts[0] = imagePrefix
```

Comment:
In this case we end up with the same value (as the third parameter in `getImagePrefix` is with the biggest priority).
Can't we skip the replacement with the same value in that case?

Reply:
Yes, that would be redundant in that case but I thought there wasn't need for special handling of it since it would ultimately be the same, just with more code